### PR TITLE
fix calculation of pg used_bytes

### PR DESCRIPTION
**Pull request by JacksonYao287**
#172

### CODE REVIEW

File: src/lib/homestore_backend/hs_homeobject.hpp [hunk 1]
```diff
@@ -175,10 +175,14 @@ class HSHomeObject : public HomeObjectImpl {
                              pg_.durable_entities().active_blob_count.load(std::memory_order_relaxed));
                 GAUGE_UPDATE(*this, tombstone_blob_count,
                              pg_.durable_entities().tombstone_blob_count.load(std::memory_order_relaxed));
+                GAUGE_UPDATE(*this, total_occupied_space,
+                             pg_.durable_entities().total_occupied_blk_count.load(std::memory_order_relaxed) *
+                                 blk_size);
```
Comment (yuwmao) [inline src/lib/homestore_backend/hs_homeobject.hpp:180]: When will blk_size be updated?

Comment (JacksonYao287) [reply src/lib/homestore_backend/hs_homeobject.hpp:180]: blk_size is a fixed value, it will not be updated after initialized

Comment (yuwmao) [reply src/lib/homestore_backend/hs_homeobject.hpp:180]: Sorry, I read it wrong.

File: src/lib/homestore_backend/hs_shard_manager.cpp [hunk 2]
```diff
@@ -328,6 +328,22 @@ void HSHomeObject::on_shard_message_commit(int64_t lsn, sisl::blob const& h, hom
             chunk_selector_->select_specific_chunk(blkids.chunk_num());
         }
         if (ctx) { ctx->promise_.setValue(ShardManager::Result< ShardInfo >(shard_info)); }
+
+        // TODO: add shard header blk count to total_occupied_blk_count and handle the case of replaying journal after
```
Comment (xiaoxichen) [inline src/lib/homestore_backend/hs_shard_manager.cpp:332]: if this is the case, it also involve  in on_blob_put_commit?

Comment (JacksonYao287) [reply src/lib/homestore_backend/hs_shard_manager.cpp:332]: yes, all the action in `on_commit` will be involved. we need to revisit this later.

Comment (xiaoxichen) [reply src/lib/homestore_backend/hs_shard_manager.cpp:332]: how other metrics handle this case? IIRC we restart from a checkpoint ?

Comment (JacksonYao287) [reply src/lib/homestore_backend/hs_shard_manager.cpp:332]: i believe other metrics do not handle this case. as you said,  we definitely restart from a checkpoint. but in homestore, checkpoint is not a real checkpoint, partial persistence might happen. actually, when we do cp_flush, we just start several future to flush those cp of every component. the key point here is that no atomicity of the cp of every component is promised. this is the potential risk of inconsistence。

Comment (JacksonYao287) [reply src/lib/homestore_backend/hs_shard_manager.cpp:332]: I have post this question in homestore channel. hair will fix it.
https://ebay-eng.slack.com/archives/CAVQXEAKS/p1714099698282209

**Review by xiaoxichen — APPROVED** [review]

**Pull request merged**